Source/JavaWrappers.cpp

```diff
@@ -274,6 +281,88 @@ namespace java::lang
     }
 }
 
+namespace java::websocket
+{
+    WebSocketClient::WebSocketClient(std::string url, std::function<void(std::string)> message, std::function<void()> open, std::function<void()> close, std::function<void()> error)
+        : Object{android::global::GetWebSocketClass()}
+        , m_openCallback{std::move(open)}
+        , m_messageCallback{std::move(message)}
+        , m_closeCallback{std::move(close)}
+        , m_errorCallback{std::move(error)}
+    {
+        static JNINativeMethod methods[] = {
+                {"closeCallback", "()V", (void*)OnClose },
+                {"openCallback", "()V", (void*)&WebSocketClient::OnOpen },
+                {"messageCallback", "(Ljava/lang/String;)V", (void*)&WebSocketClient::OnMessage },
+                {"errorCallback", "()V", (void*)&WebSocketClient::OnError },
+        };
+        m_env->RegisterNatives(m_class, methods, 4);
+
+        // initialize socket
+        JObject(m_env->NewObject(m_class, m_env->GetMethodID(m_class, "<init>", "(Ljava/lang/String;)V"), m_env->NewStringUTF(url.c_str())));
+
+        WebSocketClient *p_webSocketClient = this;
+        s_objectMap.insert({JObject(), p_webSocketClient});
```

Comment:
>p_webSocketClient [](http://example.com/codeflow?start=39&length=17)

why not just pass this pointer?

Reply:
+1

Reply:
Doing `s_objectMap.insert({JObject(), &this});`  gave me the error `error: cannot take the address of an rvalue of type 'java::websocket::WebSocketClient *'`.

Reply:
don't use &this, "this" keyword is already a pointer.